### PR TITLE
tsp, cadl-ranch add test case for ARM

### PR DESCRIPTION
**Pull request by v-hongli1**
Add test cases for tsp, cadl-ranch add test case for ARM

### CODE REVIEW

File: typespec-tests/src/test/java/com/azure/arm/models/resources/ArmResourceTest.java [hunk 1]
```diff
@@ -3,20 +3,164 @@
...
-public class ArmResourceTest {
+import java.util.List;
+import java.util.stream.Collectors;
 
+public class ArmResourceTest {
+    private static final String TOP_LEVEL_TRACKED_RESOURCE_ID = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.Arm.Models.Resources/topLevelTrackedResources/top";
+    private static final String TOP_LEVEL_TRACKED_RESOURCE_NAME = "top";
+    private static final String TOP_LEVEL_TRACKED_RESOURCE_TYPE = "topLevel";
+    private static final String NESTED_PROXY_RESOURCE_ID = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.Arm.Models.Resources/topLevelTrackedResources/top/nestedProxyResources/nested";
+    private static final String NESTED_PROXY_RESOURCE_NAME_TYPE = "nested";
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/azure/arm/models/resources/ArmResourceTest.java:27]: separate this 2, NAME be NAME, TYPE be TYPE

File: typespec-tests/src/test/java/com/azure/arm/models/resources/ArmResourceTest.java [hunk 2]
```diff
@@ -3,20 +3,164 @@
...
+        TopLevelTrackedResource  topLevelTrackedResource = manager.topLevelTrackedResources()
+                .define(TOP_LEVEL_TRACKED_RESOURCE_NAME)
+                .withRegion(Region.US_EAST)
+                .withExistingResourceGroup(RESOURCE_GROUP_NAME)
+                .withProperties(new TopLevelTrackedResourceProperties().withDescription(RESOURCE_DESCRIPTION_VALID))
+                .create();
+        Assertions.assertEquals(TOP_LEVEL_TRACKED_RESOURCE_ID, topLevelTrackedResource.id());
+        Assertions.assertEquals(TOP_LEVEL_TRACKED_RESOURCE_NAME, topLevelTrackedResource.name());
+        Assertions.assertEquals(TOP_LEVEL_TRACKED_RESOURCE_TYPE, topLevelTrackedResource.type());
+        Assertions.assertEquals(TOP_LEVEL_TRACKED_RESOURCE_LOCATION, topLevelTrackedResource.location());
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/azure/arm/models/resources/ArmResourceTest.java:48]: Please compare `topLevelTrackedResource.region()`

File: typespec-tests/src/test/java/com/azure/arm/models/resources/ArmResourceTest.java [hunk 3]
```diff
@@ -3,20 +3,164 @@
...
+        });
+
+        // TopLevelTrackedResources.listBySubscription
+        topLevelTrackedResourceList = manager.topLevelTrackedResources()
+                .list()
+                .stream()
+                .filter(resource -> TOP_LEVEL_TRACKED_RESOURCE_ID.equals(resource.id()))
+                .collect(Collectors.toList());
+        Assertions.assertEquals(1, topLevelTrackedResourceList.size());
+        topLevelTrackedResourceList.forEach(resource -> {
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/azure/arm/models/resources/ArmResourceTest.java:74]: Just assert the item. Don't do forEach.

Comment (v-hongli1) [reply typespec-tests/src/test/java/com/azure/arm/models/resources/ArmResourceTest.java]: Fixed in the new version.

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**